### DISCUSSION TITLE
Out of 100 titles, only 37 are being identified.

### DISCUSSION THREAD

**Discussion by mizifih** [a1ex4/ownfoil#131] [Q&A] [answered]
I'm using Nsx Library Manager to manage my library identifying Games, Updates and DLCs. The numbers I'm using are based on Nsx Library Manager identification and organization, there isn't a single unidentified file in my library.

But, Ownfoil isn't recognizing all of them.

It was able to identify:
- 37 games out of 100
- 208 DLCs out of 235
- Can't list Updates in Ownfoil

What am I'm missing? Can it be something on my naming schemes causing this difference in discoverability? All files are named following the same patter, also, I'm adding Title ID, Patch ID and Version to the file name when possible.

I have NSP, NSZ, XCI and XCZ files. These is are the naming schemes I'm using:

- **GAME:** `[BASE]{TitleName}.TID-[{TitleId}].PID-[{PatchId}].{Extension}`
  - [BASE]Cat Quest III.TID-[010088501B8F2000].PID-[010088501B8F2800].nsp
- **UPDATE:** `[UPD]{TitleName}.TID-[{TitleId}].VER-[{Version}].{Extension}`
  - [UPD]Bayonetta 3.TID-[01004A4010FEA800].VER-[65536].nsp
- **DLC:** `[DLC]{AppName}.{TitleName}.TID-[{TitleId}].{Extension}`
  - [DLC]Super Smash Bros. Ultimate.【Costume】Erdrick's Armor and Helmet.TID-[01006A800016F025].nsp

**Comment by mizifih** [comment]
**FILES:**
```
[BASE]Cat Quest III.TID-[010088501B8F2000].PID-[010088501B8F2800].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[393216].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[131072].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[327680].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[196608].nsp
```
![image](https://github.com/user-attachments/assets/0ca077b6-616a-4f3b-824e-aa02280cebb2)

**LOG (search for "_cat_":**
```
[2024-10-16 14:02:58.101] WARNING (library) File contains None value, it will be skipped: {'id': 511, 'filepath': '/games/Nintendo - Switch/[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[327680].nsp', 'library': '/games/Nintendo - Switch', 'folder': '', 'filename': '[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[327680].nsp', 'title_id': '010088501B8F2000', 'app_id': '010088501B8F2800', 'type': 'UPDATE', 'version': None, 'extension': 'nsp', 'size': 304101144, 'identification': 'filename'}

[2024-10-16 14:02:58.163] WARNING (library) File contains None value, it will be skipped: {'id': 514, 'filepath': '/games/Nintendo - Switch/[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[196608].nsp', 'library': '/games/Nintendo - Switch', 'folder': '', 'filename': '[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[196608].nsp', 'title_id': '010088501B8F2000', 'app_id': '010088501B8F2800', 'type': 'UPDATE', 'version': None, 'extension': 'nsp', 'size': 303154968, 'identification': 'filename'}
```

**Comment by a1ex4** [comment]
Custom naming schemes are not supported currently, you should configure your console keys in the settings so content will be identified from the data in the files.

**Reply by mizifih** [reply]
The `prod.keys` file looked properly set, I've uploaded it and the Settings page says `Keys are valid!` under Console keys. I'll manually add it again and see what happens. But I used the same `prod.keys` I'm currently using with Nsx Library Manager.

I deleted `keys.txt`, redid the upload process, the Settings page says `Keys are valid!` again, but still, files aren't being recognized.

One exemple, look at this Hogwarts Legacy DLC, log says Ownfoil will skip this file, but it recognized it's `title_id`, `app_id` and `type`, how, I have no idea, maybe by filename contents, but as you said, custom naming schemes are not supported yet. 

### LOG:
`2024-10-17T13:03:18.635859298Z [2024-10-17 13:03:18.635] WARNING (library) File contains None value, it will be skipped: {'id': 535, 'filepath': '/games/Nintendo - Switch/[DLC]Hogwarts Legacy.Hogwarts Legacy: Portuguese (Brazil) Language Pack.TID-[0100F7E00C70F00C].nsp', 'library': '/games/Nintendo - Switch', 'folder': '', 'filename': '[DLC]Hogwarts Legacy.Hogwarts Legacy: Portuguese (Brazil) Language Pack.TID-[0100F7E00C70F00C].nsp', 'title_id': '0100F7E00C70E000', 'app_id': '0100F7E00C70F00C', 'type': 'DLC', 'version': None, 'extension': 'nsp', 'size': 1404574881, 'identification': 'filename'}`

**Comment by mizifih** [comment, marked as answer]
Ok, i did nothing to Ownfoil itself, but I've added some more games to the library using Nsx Library Manager and, somehow, looks like all games are now being listed in Ownfoil GUI.

So, problem solved, I guess. I'm closing this issue.
